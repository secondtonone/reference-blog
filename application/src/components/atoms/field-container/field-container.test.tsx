import renderer from 'react-test-renderer';
import FieldContainer from './field-container';

jest.mock('../tooltip', () => ({ children, error }) => (
  <span>
    {error && <span>{error}</span>}
    {children}
  </span>
));

describe('Match snapshots Field Container', () => {
  it('internal', () => {
    const tree = renderer.create(<FieldContainer
      internal
      field={() => (
        <div className="SOutline">
          <input />
        </div>
      )}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with error', () => {
    const tree = renderer.create(<FieldContainer
      error="Error"
      field={() => (
        <div className="SOutline">
          <input />
        </div>
      )}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
