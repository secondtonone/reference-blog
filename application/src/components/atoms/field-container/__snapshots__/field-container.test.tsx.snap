// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots Field Container internal 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 trAty"
  data-test="field-container"
>
  <span>
    <div
      className="SOutline"
    >
      <input />
    </div>
  </span>
</span>
`;

exports[`Match snapshots Field Container with error 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <div
      className="SOutline"
    >
      <input />
    </div>
  </span>
</span>
`;
